core/src/main/java/org/elasticsearch/http/netty/HttpUtils.java

```diff
@@ -0,0 +1,126 @@
+import static org.jboss.netty.handler.codec.http.HttpHeaders.Values.KEEP_ALIVE;
+import static org.jboss.netty.handler.codec.http.HttpMethod.CONNECT;
+import static org.jboss.netty.handler.codec.http.HttpMethod.DELETE;
+import static org.jboss.netty.handler.codec.http.HttpMethod.GET;
+import static org.jboss.netty.handler.codec.http.HttpMethod.HEAD;
+import static org.jboss.netty.handler.codec.http.HttpMethod.OPTIONS;
+import static org.jboss.netty.handler.codec.http.HttpMethod.PATCH;
+import static org.jboss.netty.handler.codec.http.HttpMethod.POST;
+import static org.jboss.netty.handler.codec.http.HttpMethod.PUT;
+import static org.jboss.netty.handler.codec.http.HttpMethod.TRACE;
+
+/**
+ * Some utils for working with Netty's Http functionality.
+ */
+public class HttpUtils {
+
+    /**
+     * Determine if the request protocol version is HTTP 1.0
+     *
+     * @param request the {@link HttpRequest}
+     * @return {@code true} if the protocol version is HTTP 1.0, otherwise {@code false}
+     */
+    public static boolean isHttp10(HttpRequest request) {
+        return request.getProtocolVersion().equals(HttpVersion.HTTP_1_0);
+    }
+
+    /**
+     * Determine if the request connection should be closed on completion.
+     *
+     * @param request the {@link HttpRequest}
+     * @return {@code true} if the connection should be closed, otherwise {@code false}
+     */
+    public static boolean isCloseConnection(HttpRequest request) {
+        final boolean http10 = isHttp10(request);
+        return CLOSE.equalsIgnoreCase(request.headers().get(CONNECTION)) ||
+            (http10 && !KEEP_ALIVE.equalsIgnoreCase(request.headers().get(CONNECTION)));
+    }
+
+    /**
+     * Create a new {@link HttpResponse} to transmit the response for the given request.
+     *
+     * @param request the {@link HttpRequest} to create a response for
+     * @return the {@link HttpResponse} object
+     */
+    public static HttpResponse newResponse(HttpRequest request) {
+        final boolean http10 = isHttp10(request);
+        final boolean close = isCloseConnection(request);
+        // Build the response object.
+        HttpResponseStatus status = HttpResponseStatus.OK; // default to initialize
+        org.jboss.netty.handler.codec.http.HttpResponse resp;
+        if (http10) {
+            resp = new DefaultHttpResponse(HttpVersion.HTTP_1_0, status);
+            if (!close) {
+                resp.headers().add(CONNECTION, "Keep-Alive");
+            }
+        } else {
+            resp = new DefaultHttpResponse(HttpVersion.HTTP_1_1, status);
+        }
+        return resp;
+    }
+
+    /**
+     * Convert a string representation of an HTTP method to its {@link HttpMethod} equivalent.
+     *
+     * @param method the method name to convert
+     * @return an {@link Optional} that contains the HttpMethod, or empty if no match was found.
+     */
+    public static Optional<HttpMethod> toHttpMethod(final String method) {
```

Comment:
This is a good point - I can get rid of the enLowerCase() anyway and toUpperCase() the input if I want case insensitivity.